android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java

```
@@ -101,6 +101,29 @@ protected Void doInBackground(final Void ... params) {
         myAsyncTask.executeOnExecutor(AsyncTask.SERIAL_EXECUTOR);
     }
 
+    @ReactMethod
+    public void queryContacts(final ReadableMap contact, final Promise promise) {
+        int pageSize = contact.hasKey("limit") ? contact.getInt("limit") : 10000;
+        int pageOffset = contact.hasKey("offset") ? contact.getInt("offset") : 0;
+        String searchTerm = contact.hasKey("searchTerm") ? contact.getString("searchTerm") : null;
+
+        AsyncTask<Void,Void,Void> myAsyncTask = new AsyncTask<Void,Void,Void>() {
+            @Override
+            protected Void doInBackground(final Void ... params) {
+                Context context = getReactApplicationContext();
+                ContentResolver cr = context.getContentResolver();
+
+                ContactsProvider contactsProvider = new ContactsProvider(cr);
+                WritableArray contacts = contactsProvider.queryContacts(searchTerm, pageOffset, pageSize);
+                promise.resolve(contacts);
```

will this bubble up the actual error to JS? It would be good if we can also surface the error that's happening in case it's not just pagination

---

fixed